### PR TITLE
Mdc migration2

### PR DESCRIPTION
**Pull request by HejdaJakub**
* In this PR were migrated to MDC these components: form-field, input, list, menu, paginator,progress-bar (does not exist in current GUI), progress-spinner (does not exist in current GUI).
* Mat-label is now the required part of the form-field (placeholders is no longer a valid substitution of the floating label), so mat-labels were added to all missing places. Dynamic sizing of space for hint and error under the form-filed has to be defined explicitly.
* Outlined mat-form-field was partly customized (oulined mat-form-filed in tables) to form-filed without border.
* HTML tag <input> with matInput attribute have to be inside <mat-form-field> now.
* Some small changes were applied also to list components to fix the design.
* Our global custom classes for progress-bar were removed cause this component is not used in out GUI.

### CODE REVIEW

File: apps/user-profile/src/app/pages/settings-page/settings-preferred-shells/settings-preferred-shells.component.html [hunk 1]
```diff
@@ -14,20 +14,22 @@ <h1 class="page-title">{{'PREFERRED_SHELLS.TITLE' | customTranslate | translate}
...
-          *ngFor="let defShell of defaultShells"
-          (click)="changeValue()"
-          [value]="defShell"
-          >{{defShell}}</mat-option
-        >
-      </mat-select>
-    </mat-form-field>
+  <mat-list-item *ngFor="let shell of shells; let i = index" class="mb-2">
+    <div class="d-flex flex-row">
+      <mat-form-field class="w-100">
```
Comment (bodnara) [inline apps/user-profile/src/app/pages/settings-page/settings-preferred-shells/settings-preferred-shells.component.html:19]: Does this need to have `w-100` ?

File: apps/admin-gui/src/styles.scss [hunk 2]
```diff
@@ -1689,3 +1676,64 @@ td.mat-mdc-cell {
...
+mat-form-field mat-icon {
+  padding: 0 !important;
+}
+// TODO: Do we want to have a transparent background for focused elements?
+//.mat-focused .mat-mdc-form-field-focus-overlay {
+//  background-color: transparent !important;
+//}
+// TODO: Do we want to have left padding for labels and text in form field? (the same for hints and errors)?
+// Customization for "Items per page" form field
+//.mdc-text-field:not(.mdc-text-field--outlined) {
```
Comment (bodnara) [inline apps/admin-gui/src/styles.scss:1710]: Not sure if we need additional padding for the labels. It seems there is already some left margin.

File: apps/user-profile/src/app/pages/settings-page/settings-preferred-shells/settings-preferred-shells.component.html (hunk 1 above)
Comment (HejdaJakub) [reply apps/user-profile/src/app/pages/settings-page/settings-preferred-shells/settings-preferred-shells.component.html:19]: I think it's not necessary.

File: apps/publications/src/app/dialogs/add-category-dialog/add-category-dialog.component.html [hunk 3]
```diff
@@ -4,21 +4,15 @@ <h1 mat-dialog-title>{{"DIALOGS.ADD_CATEGORY.TITLE" | translate}}</h1>
...
         </mat-error>
       </mat-form-field>
       <mat-form-field>
-        <input
-          matInput
-          required
-          placeholder="{{'DIALOGS.ADD_CATEGORY.RANK' | translate}}"
-          [formControl]="rankCtrl" />
+        <mat-label>{{'DIALOGS.ADD_CATEGORY.RANK' | translate}}</mat-label>
+        <input matInput required [formControl]="rankCtrl" />
```
[the remark this answers is not in the document]
Comment (HejdaJakub) [reply apps/publications/src/app/dialogs/add-category-dialog/add-category-dialog.component.html:15]: I moved it into the mat-hint inside the mat-form-field.

File: apps/publications/src/styles.scss [hunk 4]
```diff
@@ -351,3 +353,28 @@ mat-icon {
...
+  padding-top: 16px !important;
+  width: 100%;
+}
+.mat-form-field-appearance-outline {
+  label:not(.mdc-floating-label--float-above) {
+    padding-top: 0 !important;
+  }
+}
+.mdc-floating-label--float-above {
+  width: 133.33% !important;
```
[the remark this answers is not in the document]
Comment (HejdaJakub) [reply apps/publications/src/styles.scss:376]: Firefox got me again :( ... now it is fixed.

File: apps/admin-gui/src/styles.scss (hunk 2 above)
Comment (HejdaJakub) [reply apps/admin-gui/src/styles.scss:1710]: There is actually left padding 16px and this class instead has the aim to replace this "big" padding with just a small 2px left padding. But in my opinion, the actual padding is ok and it was just a suggestion, so I am removing the comment.

**Comment by HejdaJakub** [comment]
Global styles for admin-gui changed (to adjust new multiline string attribute).

**Review by bodnara — APPROVED** [review]
Looks good. I would lean towards merging this even with just one review so rest of the tasks can adjust to changes. What do you think ?

**Review by xflord — COMMENTED** [review]
Screenshot from 2023-02-06 10-27-15
I'm not a big fan of this hint alignment, otherwise looks good

**Pull request merged**